redbot/core/_global_checks.py

```diff
@@ -5,4 +5,6 @@
 def init_global_checks(bot):
     @bot.check_once
     async def check_message_is_eligible_as_command(ctx: commands.Context) -> bool:
+        if ctx.interaction is not None:
+            return True
```

Comment:
I think we should add a comment here and possibly also in `Red.message_eligible_as_command` so that we don't introduce discrepancies when changing the logic in the future:
```suggestion
        if ctx.interaction is not None:
            # equivalent checks are performed by `RedTree.interaction_check`
            return True
```